include/beast/http/impl/basic_parser_v1.ipp

```diff
@@ -913,9 +989,9 @@ write(boost::asio::const_buffer const& buffer, error_code& ec)
                 flags_ |= parse_flag::skipbody;
                 break;
             case body_what::pause:
-                s_ = s_headers_done;
                 return used();
             }
+            --p;
```

Comment:
They are necessary. It has to do with how the parser transitions state and when it consumes a character (by executing `break`) versus when it changes state without advancing the input pointer (by executing `goto redo`). Worth studying to make sure I didn't screw it up...